tiledb/sm/filesystem/gcs.cc

```diff
@@ -859,9 +860,21 @@ Status GCS::flush_object(const URI& uri) {
     source_objects.emplace_back(std::move(source_object));
   }
 
+  // Provide a best-effort attempt to delete any stale, intermediate
+  // composed objects.
+  const std::string prefix = object_path + "__compose";
```

Comment:
Should we use [CreateRandomPrefixName](https://googleapis.dev/cpp/google-cloud-storage/latest/namespacegoogle_1_1cloud_1_1storage_1_1v1.html#a7d4aa7386c7b28939c59663951361d76)?
(per the [docs](https://googleapis.dev/cpp/google-cloud-storage/latest/namespacegoogle_1_1cloud_1_1storage_1_1v1.html#a848d7fcfc556c5e45fc552ea81e68054)) or is `object_path` always unique enough here?

Reply:
The intent here is that object_path is sufficiently unique between different objects but identical for retries on the same object so that we will overwrite stale intermediate objects in the event of a retry-after-failure